File: Week_09/Seminar/Solutions/kth_largest_element/solution_pq.py
from queue import PriorityQueue

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        pq = PriorityQueue()

        for num in nums:
            pq.put(num)

        p = len(nums) - k

        for _ in range(p):
            pq.get()

        return pq.queue[0]

File: Week_10/Seminar/Solutions/KthLargestElement/solution_pq.cpp
#include <queue>

using namespace std;

class Solution
{
public:
    int findKthLargest(vector<int> &nums, int k)
    {
        std::priority_queue<int> pq;

        for (int el : nums)
        {
            pq.push(el);
        }

        for (int i = 1; i < k; i++)
        {
            pq.pop();
        }

        return pq.top();
    }
};